### PR TITLE
Adding more user documentation

### PR DESCRIPTION
**Pull request by Bento007**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -99,39 +99,99 @@ If you're ok with the changes run `make deploy-infra`.
...
-**DEPLOYMENT** - used to set the current deployment of fusillade to target. This determines what deployment variables to
+- **DEPLOYMENT** - used to set the current deployment of fusillade to target. This determines what deployment 
+variables to
  source from `environment`. 
-**GITHUB_TOKEN_PATH** - Point to the location of a file in your local directory containing a github token used for 
+- **GITHUB_TOKEN_PATH** - Point to the location of a file in your local directory containing a github token used for 
  promoting fusillade branches and publishing new version. If GITHUB_TOKEN_SECRET_NAME is also present, GITHUB_TOKEN_PATH
  take precedence over GITHUB_TOKEN_SECRET_NAME.
-**GITHUB_TOKEN_SECRET_NAME** - oint to the location of a AWS parameters key containing a github token used for 
+- **GITHUB_TOKEN_SECRET_NAME** - oint to the location of a AWS parameters key containing a github token used for
```
Comment (amarjandu) [inline Readme.md]: typo

Comment (Bento007) [reply Readme.md]: fixed

Comment (DailyDreaming) [inline Readme.md]: `oint to the location`

File: Readme.md [hunk 2]
```diff
@@ -99,39 +99,99 @@ If you're ok with the changes run `make deploy-infra`.
...
+**Note:** New *fusillade_admins* can be assigned using the fusillade API and assigning the role of *fusillade_admin* 
+to a user.
+
+## Users
+A user can represent a service account, or a personal account. They can be explicitly created or created on demand when
+a user's permissions are first evaluated. A user is automatically added to `/group/user_default` when created. All other
+roles and groups must be added using the fusillade API.
+
+## Roles
+Roles contains policies that are used to determine what a user can do. A role can either be directly applied to a
```
[the remark this answers is not in the document]
Comment (Bento007) [reply Readme.md]: fixed

File: Readme.md [hunk 3]
```diff
@@ -99,39 +99,99 @@ If you're ok with the changes run `make deploy-infra`.
...
+ 
+When the permissions of a user is evaluated, all policies attached to a user, the user's groups, and the user's are 
+used.
+  
+### Defining Policy
+Uses [AWS IAM Policy grammar](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html) to 
+define your services permissions.
+For resource names use the same format as [AWS Service NameSpace](https://docs.aws.amazon
+.com/general/latest/gr/aws-arns-and-namespaces
+.html).
```
[the remark this answers is not in the document]
Comment (Bento007) [reply Readme.md]: fixed

File: Readme.md [hunk 4]
```diff
@@ -99,39 +99,99 @@ If you're ok with the changes run `make deploy-infra`.
...
+
+### Special Fusillade Context Keys for Policies
+
+In the same way AWS iam provides [context keys available to all services](https://docs.aws.amazon
```
[the remark this answers is not in the document]
Comment (Bento007) [reply Readme.md]: fixed

**Pull request merged**